crates/iceberg-catalog/src/catalog/config.rs

```diff
@@ -142,6 +133,30 @@ impl<
     }
 }
 
+async fn maybe_add_user<D: Catalog>(
+    request_metadata: &RequestMetadata,
+    state: <D as Catalog>::State,
+) -> Result<()> {
+    if let Some(user_id) = request_metadata.user_id() {
+        // If the user is authenticated, create a user in the catalog
+        let user = D::create_user(
+            user_id,
+            request_metadata.user_display_name(),
+            request_metadata.user_name().as_deref(),
+            request_metadata.email(),
+            UserOrigin::ImplicitRegistration("config".to_string()),
```

Comment:
maybe change to uri of request or remove the inner string instead, would make the db side of things cleaner, currently we're getting a text column instead of an enum there.